frontend/packages/@depmap/dataset-manager/src/components/DimensionTypeForm.tsx

```diff
@@ -31,9 +30,12 @@ export default function DimensionTypeForm(props: DimensionTypeFormProps) {
 
   React.useEffect(() => {
     if (isEditMode && dimensionTypeToEdit) {
-      const datasetsWithDimensionType: TabularDataset[] = datasets.filter(
+      const publicDatasetsWithDimensionType: TabularDataset[] = datasets.filter(
         (d) => {
-          return d.index_type_name === dimensionTypeToEdit.name;
+          return (
+            d.index_type_name === dimensionTypeToEdit.name &&
+            d.group.name === "Public"
```

Comment:
If we're going to hardcode something, I think it should be the ID not the label. In the backend we have a constant for the public group ID for that purpose